integrations/jira/main.py

```diff
@@ -54,9 +60,41 @@ async def on_resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
         ocean.integration_config["atlassian_user_token"],
     )
 
-    async for issues in client.get_paginated_issues():
-        logger.info(f"Received issue batch with {len(issues)} issues")
-        yield issues
+    async for boards in client.get_boards():
+        logger.info(f"Received board batch with {len(boards)} boards")
+        for board in boards:
+            async for issues in client.get_issues(board["id"]):
+                logger.info(f"Received issue batch with {len(issues)} issues")
+                yield issues
+
+
+@ocean.on_resync(ObjectKind.BOARD)
+async def on_resync_boards(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    client = JiraClient(
+        ocean.integration_config["jira_host"],
+        ocean.integration_config["atlassian_user_email"],
+        ocean.integration_config["atlassian_user_token"],
+    )
```

Comment:
You can create a function that will initialize and return the jira client, and reuse it. This will save you from initializing it in each resync method. something like this from the opsgenie integration

def init_client() -> OpsGenieClient:
    return OpsGenieClient(
        ocean.integration_config["api_token"],
        ocean.integration_config["api_url"],
    )